### DISCUSSION TITLE
problem installing pyxcp

### DISCUSSION THREAD

**Discussion by MCO-SOCOMEC** [christoph2/pyxcp#173] [Q&A] [answered]
Hi,
I'm trying to install pyxcp but during the installation process I get an error. Could you give me any insight? I'm not very skilled with python. I'm usinng Python 3.9.6.
thanks
Michele

package Upgrade/Installation
Collecting pyxcp
  Downloading pyxcp-0.22.5.tar.gz (213 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting bandit<2.0.0,>=1.7.8 (from pyxcp)
  Downloading bandit-1.7.10-py3-none-any.whl.metadata (6.7 kB)
Collecting chardet<6.0.0,>=5.2.0 (from pyxcp)
  Downloading chardet-5.2.0-py3-none-any.whl.metadata (3.4 kB)
Collecting construct<3.0.0,>=2.10.68 (from pyxcp)
  Downloading construct-2.10.70-py3-none-any.whl.metadata (4.2 kB)
Collecting line-profiler-pycharm<2.0.0,>=1.1.0 (from pyxcp)
  Downloading line_profiler_pycharm-1.2.0-py3-none-any.whl.metadata (1.6 kB)
Requirement already satisfied: mako<2.0.0,>=1.2.4 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp) (1.3.5)
Requirement already satisfied: pyserial<4.0,>=3.5 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp) (3.5)
Collecting python-can<5.0.0,>=4.2.2 (from pyxcp)
  Downloading python_can-4.4.2-py3-none-any.whl.metadata (8.8 kB)
Collecting pytz<2025.0,>=2024.1 (from pyxcp)
  Downloading pytz-2024.2-py2.py3-none-any.whl.metadata (22 kB)
Collecting pyusb<2.0.0,>=1.2.1 (from pyxcp)
  Downloading pyusb-1.2.1-py3-none-any.whl.metadata (2.2 kB)
Collecting rich<14.0.0,>=13.6.0 (from pyxcp)
  Downloading rich-13.9.1-py3-none-any.whl.metadata (18 kB)
Requirement already satisfied: toml<0.11.0,>=0.10.2 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp) (0.10.2)
Collecting tomlkit<0.13.0,>=0.12.5 (from pyxcp)
  Downloading tomlkit-0.12.5-py3-none-any.whl.metadata (2.7 kB)
Requirement already satisfied: traitlets<=5.11.2 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp) (5.4.0)
Collecting uptime<4.0.0,>=3.0.1 (from pyxcp)
  Downloading uptime-3.0.1.tar.gz (6.6 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting PyYAML>=5.3.1 (from bandit<2.0.0,>=1.7.8->pyxcp)
  Downloading PyYAML-6.0.2-cp39-cp39-win_amd64.whl.metadata (2.1 kB)
Collecting stevedore>=1.20.0 (from bandit<2.0.0,>=1.7.8->pyxcp)
  Downloading stevedore-5.3.0-py3-none-any.whl.metadata (2.3 kB)
Requirement already satisfied: colorama>=0.3.9 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from bandit<2.0.0,>=1.7.8->pyxcp) (0.4.6)
Collecting line-profiler (from line-profiler-pycharm<2.0.0,>=1.1.0->pyxcp)
  Downloading line_profiler-4.1.3-cp39-cp39-win_amd64.whl.metadata (35 kB)
Requirement already satisfied: MarkupSafe>=0.9.2 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from mako<2.0.0,>=1.2.4->pyxcp) (2.1.1)
Requirement already satisfied: wrapt~=1.10 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp) (1.12.1)
Requirement already satisfied: packaging>=23.1 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp) (23.2)
Requirement already satisfied: typing-extensions>=3.10.0.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp) (4.8.0)
Requirement already satisfied: pywin32>=305 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp) (305)
Collecting markdown-it-py>=2.2.0 (from rich<14.0.0,>=13.6.0->pyxcp)
  Downloading markdown_it_py-3.0.0-py3-none-any.whl.metadata (6.9 kB)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from rich<14.0.0,>=13.6.0->pyxcp) (2.16.1)
Collecting mdurl~=0.1 (from markdown-it-py>=2.2.0->rich<14.0.0,>=13.6.0->pyxcp)
  Downloading mdurl-0.1.2-py3-none-any.whl.metadata (1.6 kB)
Collecting pbr>=2.0.0 (from stevedore>=1.20.0->bandit<2.0.0,>=1.7.8->pyxcp)
  Downloading pbr-6.1.0-py2.py3-none-any.whl.metadata (3.4 kB)
Downloading bandit-1.7.10-py3-none-any.whl (130 kB)
Downloading chardet-5.2.0-py3-none-any.whl (199 kB)
Downloading construct-2.10.70-py3-none-any.whl (63 kB)
Downloading line_profiler_pycharm-1.2.0-py3-none-any.whl (4.8 kB)
Downloading python_can-4.4.2-py3-none-any.whl (265 kB)
Downloading pytz-2024.2-py2.py3-none-any.whl (508 kB)
Downloading pyusb-1.2.1-py3-none-any.whl (58 kB)
Downloading rich-13.9.1-py3-none-any.whl (242 kB)
Downloading tomlkit-0.12.5-py3-none-any.whl (37 kB)
Downloading markdown_it_py-3.0.0-py3-none-any.whl (87 kB)
Downloading PyYAML-6.0.2-cp39-cp39-win_amd64.whl (162 kB)
Downloading stevedore-5.3.0-py3-none-any.whl (49 kB)
Downloading line_profiler-4.1.3-cp39-cp39-win_amd64.whl (125 kB)
Downloading mdurl-0.1.2-py3-none-any.whl (10.0 kB)
Downloading pbr-6.1.0-py2.py3-none-any.whl (108 kB)
Building wheels for collected packages: pyxcp, uptime
  Building wheel for pyxcp (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pyxcp (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [38 lines of output]
      Platform Windows
      build_ext::build_extension()
       BUILD-TYPE: 'Release'
      ================================================================================
                                     Step #1: Configure
      ================================================================================
      Traceback (most recent call last):
        File "C:\Users\MCO\AppData\Local\Temp\pip-install-dwqx_xfs\pyxcp_3897fa7794824f0f902f2eb80fe1322e\build_ext.py", line 147, in <module>
          build_extension(False)
        File "C:\Users\MCO\AppData\Local\Temp\pip-install-dwqx_xfs\pyxcp_3897fa7794824f0f902f2eb80fe1322e\build_ext.py", line 131, in build_extension
          subprocess.run(["cmake", "-S", str(TOP_DIR), *cmake_args], cwd=build_temp, check=True)  # nosec
        File "C:\Program Files\Python\lib\subprocess.py", line 505, in run
          with Popen(*popenargs, **kwargs) as process:
        File "C:\Program Files\Python\lib\subprocess.py", line 951, in __init__
          self._execute_child(args, executable, preexec_fn, close_fds,
        File "C:\Program Files\Python\lib\subprocess.py", line 1420, in _execute_child
          hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
      FileNotFoundError: [WinError 2] Impossibile trovare il file specificato
      Traceback (most recent call last):
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-uaeypfx8\overlay\Lib\site-packages\poetry\core\masonry\api.py", line 58, in build_wheel
          return WheelBuilder.make_in(
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-uaeypfx8\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 88, in make_in
          wb.build(target_dir=directory)
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-uaeypfx8\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 123, in build
          self._build(zip_file)
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-uaeypfx8\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 172, in _build
          self._run_build_script(self._package.build_script)
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-uaeypfx8\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 262, in _run_build_script
          subprocess.check_call([self.executable.as_posix(), build_script])
        File "C:\Program Files\Python\lib\subprocess.py", line 373, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['C:/Dev/hot_upgrade/atp_core/socomecenv/socomecenv/Scripts/python.exe', 'build_ext.py']' returned non-zero exit status 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyxcp
  Building wheel for uptime (pyproject.toml) ... done
  Created wheel for uptime: filename=uptime-3.0.1-cp39-cp39-win_amd64.whl size=6375 sha256=8d535323defdbe468da15b42ba3ede2cb7037319bbeda68d0285424f616b246d
  Stored in directory: C:\Users\MCO\AppData\Local\Temp\pip-ephem-wheel-cache-cokso2ok\wheels\ff\d2\cd\e18d80e7974cad2a474c85ae336fedadc7d0b6ac7207dacf5c
Successfully built uptime
Failed to build pyxcp
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (pyxcp)

**Comment by christoph2** [comment]
Hello Michele,

The error message means, that [cmake](https://cmake.org/) is missing on your system.

Building `pyxcp` on Windows is somewhat tricky, so normally there are up-to-date prebuild packages available, but there were some breaking changes in the Github build infrastructure...

But you may resort to a slightly older version:
```pip install "pyxcp==0.22.1"```

There's also a working Docker configuration:

``` Docker
FROM ubuntu:24.04

RUN apt update
RUN apt upgrade -y
RUN apt install -y pkg-config
RUN apt install -y gcc git cmake libssl-dev python3 python3-pip python3-venv python3-poetry pipx rustc cargo libffi-dev
RUN apt install -y zsh nano tmux

CMD ["/bin/zsh"]
```

**Comment by MCO-SOCOMEC** [comment]
Hello Christoph,
thank you very much for the fast reply. I went for the first solution and it worked fine!
thanks a lot
Michele

**Comment by MCO-SOCOMEC** [comment]
Hi,
I looked at the documentation and I'm trying to understand a little bit the staff being a newbie of xcp, ..
I tried to run 
xcp-profile convert -c conf_can_user.toml -o conf_can_user.py
or 
python xcphello.py

but I always get an error 

ModuleNotFoundError: No module named 'pyxcp.cpp_ext.cpp_ext'

.. I guess something is missing. Am I wrong?
thanks
Michele

**Comment by christoph2** [comment]
Looks much like you have downloaded/forked pyXCP without building the project, otherwise the `cpp_ext` would be available.
Please try to run
```shell
pip install -e .
```
from your pyXCP installation directory.

**Reply by MCO-SOCOMEC** [reply]
I'm really sorry to bother you but normally when I install python library I don't need to do anything special.
I activate the venv and I entered in the package folder "..\venv\Lib\site-packages\pyxcp" and I launch the command you suggested by I got the following error. Am I in the wrong folder?
thanks again
Michele


(socomecenv) C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\Lib\site-packages\pyxcp>pip install -e .
Obtaining file:///C:/Dev/hot_upgrade/atp_core/socomecenv/socomecenv/Lib/site-packages/pyxcp
ERROR: file:///C:/Dev/hot_upgrade/atp_core/socomecenv/socomecenv/Lib/site-packages/pyxcp does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

**Comment by christoph2** [comment]
`site-packages`  is the target directory of Python packages, you should never do modifications here.

Try to clone pyXCP to a working directory of your choice
```shell
git clone https://github.com/christoph2/pyxcp
```
and then
```shell
pip install -e .
```
In the meantime I'm working on the Github packaging issue.

**Reply by MCO-SOCOMEC** [reply]
Hello,
I tried but I guess I'm missing something. Maybe cmake?
thanks
Michele


Obtaining file:///C:/Dev/hot_upgrade/atp_core/socomecenv/xcp/pyxcp-master
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Requirement already satisfied: bandit<2.0.0,>=1.7.8 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (1.7.10)
Requirement already satisfied: chardet<6.0.0,>=5.2.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (5.2.0)
Requirement already satisfied: construct<3.0.0,>=2.10.68 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (2.10.70)
Requirement already satisfied: line-profiler-pycharm<2.0.0,>=1.1.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (1.2.0)
Requirement already satisfied: mako<2.0.0,>=1.2.4 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (1.3.5)
Requirement already satisfied: pyserial<4.0,>=3.5 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (3.5)
Requirement already satisfied: python-can<5.0.0,>=4.2.2 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (4.4.2)
Requirement already satisfied: pytz<2025.0,>=2024.1 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (2024.2)
Requirement already satisfied: pyusb<2.0.0,>=1.2.1 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (1.2.1)
Requirement already satisfied: rich<14.0.0,>=13.6.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (13.9.1)
Requirement already satisfied: toml<0.11.0,>=0.10.2 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (0.10.2)
Requirement already satisfied: tomlkit<0.13.0,>=0.12.5 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (0.12.5)
Requirement already satisfied: traitlets<=5.11.2 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (5.4.0)
Requirement already satisfied: uptime<4.0.0,>=3.0.1 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from pyxcp==0.22.7) (3.0.1)
Requirement already satisfied: PyYAML>=5.3.1 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from bandit<2.0.0,>=1.7.8->pyxcp==0.22.7) (6.0.2)
Requirement already satisfied: stevedore>=1.20.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from bandit<2.0.0,>=1.7.8->pyxcp==0.22.7) (5.3.0)
Requirement already satisfied: colorama>=0.3.9 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from bandit<2.0.0,>=1.7.8->pyxcp==0.22.7) (0.4.6)
Requirement already satisfied: line-profiler in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from line-profiler-pycharm<2.0.0,>=1.1.0->pyxcp==0.22.7) (4.1.3)
Requirement already satisfied: MarkupSafe>=0.9.2 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from mako<2.0.0,>=1.2.4->pyxcp==0.22.7) (2.1.1)
Requirement already satisfied: wrapt~=1.10 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp==0.22.7) (1.12.1)
Requirement already satisfied: packaging>=23.1 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp==0.22.7) (23.2)
Requirement already satisfied: typing-extensions>=3.10.0.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp==0.22.7) (4.8.0)
Requirement already satisfied: pywin32>=305 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from python-can<5.0.0,>=4.2.2->pyxcp==0.22.7) (305)
Requirement already satisfied: markdown-it-py>=2.2.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from rich<14.0.0,>=13.6.0->pyxcp==0.22.7) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from rich<14.0.0,>=13.6.0->pyxcp==0.22.7) (2.16.1)
Requirement already satisfied: mdurl~=0.1 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from markdown-it-py>=2.2.0->rich<14.0.0,>=13.6.0->pyxcp==0.22.7) (0.1.2)
Requirement already satisfied: pbr>=2.0.0 in c:\dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages (from stevedore>=1.20.0->bandit<2.0.0,>=1.7.8->pyxcp==0.22.7) (6.1.0)
Building wheels for collected packages: pyxcp
  Building editable for pyxcp (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building editable for pyxcp (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [38 lines of output]
      Platform Windows
      build_ext::build_extension()
       BUILD-TYPE: 'Release'
      ================================================================================
                                     Step #1: Configure
      ================================================================================
      Traceback (most recent call last):
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\xcp\pyxcp-master\build_ext.py", line 147, in <module>
          build_extension(False)
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\xcp\pyxcp-master\build_ext.py", line 131, in build_extension
          subprocess.run(["cmake", "-S", str(TOP_DIR), *cmake_args], cwd=build_temp, check=True)  # nosec
        File "C:\Program Files\Python\lib\subprocess.py", line 505, in run
          with Popen(*popenargs, **kwargs) as process:
        File "C:\Program Files\Python\lib\subprocess.py", line 951, in __init__
          self._execute_child(args, executable, preexec_fn, close_fds,
        File "C:\Program Files\Python\lib\subprocess.py", line 1420, in _execute_child
          hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
      FileNotFoundError: [WinError 2] Impossibile trovare il file specificato
      Traceback (most recent call last):
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 273, in build_editable
          return hook(wheel_directory, config_settings, metadata_directory)
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-bp_14v9q\overlay\Lib\site-packages\poetry\core\masonry\api.py", line 82, in build_editable
          return WheelBuilder.make_in(
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-bp_14v9q\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 88, in make_in
          wb.build(target_dir=directory)
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-bp_14v9q\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 117, in build
          self._build(zip_file)
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-bp_14v9q\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 172, in _build
          self._run_build_script(self._package.build_script)
        File "C:\Users\MCO\AppData\Local\Temp\pip-build-env-bp_14v9q\overlay\Lib\site-packages\poetry\core\masonry\builders\wheel.py", line 262, in _run_build_script
          subprocess.check_call([self.executable.as_posix(), build_script])
        File "C:\Program Files\Python\lib\subprocess.py", line 373, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['C:/Dev/hot_upgrade/atp_core/socomecenv/socomecenv/Scripts/python.exe', 'build_ext.py']' returned non-zero exit status 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for pyxcp
Failed to build pyxcp
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (pyxcp)

**Comment by christoph2** [comment]
Yes, the cmake executable is still missing on your system.

**Reply by MCO-SOCOMEC** [reply]
Hi, I saw you did some commits so I tried to update to the last version. It seems to work now even if I got an error at setup of Daq processor (I don't know if it's linked to the version of Python I'm using)

File "C:\Dev\hot_upgrade\atp_core\socomecenv\socomecenv\lib\site-packages\pyxcp\daq_stim\__init__.py", line 38, in DaqProcessor
    def setup(self, start_datetime: CurrentDatetime | None = None, write_multiple: bool = True):
TypeError: unsupported operand type(s) for |: 'pybind11_type' and 'NoneType'

I fastly fixed it removing the "| None = None" and now xcp-profile convert -c conf_can_user.toml -o conf_can_user.py seems to work fine

class DaqProcessor:
    def __init__(self, daq_lists: List[DaqList]):
        # super().__init__()
        self.daq_lists = daq_lists
        self.log = get_application().log

    #def setup(self, start_datetime: CurrentDatetime | None = None, write_multiple: bool = True):
    def setup(self, start_datetime: CurrentDatetime, write_multiple: bool = True):

**Comment by christoph2** [comment, marked as answer]
Yes, the `| None` notation is relatively new and does not work before Python 3.10, I think.
The fix will be included in the next commit.

P.S.: pre-build Windows packages are now available (Python 3.9 ... 3.13), so

```shell
pip install pyxcp
```
should work.

Also note: The is now a script `xcp-examples`, which installs the example stuff to a given directory.
